### PR TITLE
#169421389: Add statistics data layer

### PR DESCRIPTION
**Pull request by vteremasov**
* Fix bugs
 * Fix tests

### CODE REVIEW

File: packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  });
+}
+
+export const StatisticsController: React.FC = () => {
+  const {
+    client,
+    config: { title },
+  } = useWidgetContext();
+  const dateNow = Date.now();
+  const oneDay = 86400000;
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx]: Can we name it in `UPPER_SNAKE` ?

File: packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  const todayStatDate = dateNow - (dateNow % oneDay) - 200;
+  const yesterdayStatDate = todayStatDate - oneDay - 200;
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx]: Why minus 200?

File: packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    rateYesterday.loading &&
+    rateToday.loading;
+
+  const error =
+    kisYesterday.error &&
+    tasksToday.error &&
+    tasksYesterday.error &&
+    tasksToday.error &&
+    rateYesterday.error &&
+    rateToday.error;
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx]: Maybe `or` between values? And what we should do on UI when 5 values will be without error, and the last one will be with error?

File: packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  });
+
+  const [rateYesterday] = useStatisticsData(client, 'AutomationRate', {
+    from: yesterdayStatDate,
+    to: todayStatDate,
+  });
+  const [rateToday] = useStatisticsData(client, 'AutomationRate', {
+    from: todayStatDate,
+    to: dateNow,
+  });
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx]: 1. Can we add limit to params ```{from: yesterdayStatDate, to: dateNow, limit: 2}``` ? And then get today and yesterday data?
2. I think we can move`{from, to}`  to variables (something like this `todayQueryParams, yesterdayQueryParams` )

File: packages/bootstrap/src/components/dashboard/hooks/useTimeseries.ts [hunk 5]
```diff
@@ -1,14 +1,25 @@
...
-  to: number;
-  limit: number;
+  from?: number;
+  to?: number;
+  limit?: number;
 }
 
 export const fetchTimeseriesVertexId = async (client: Client, type: string) => {
-  const query = `+(ogit\\/_type:"ogit/Timeseries" \\/DataName : "Dashboard.${type}")`;
+  const query = `ogit\\/_type:"ogit/Timeseries" AND \\/DataName : "Dashboard.${type}"`;
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/hooks/useTimeseries.ts]: why?

File: packages/bootstrap/src/components/dashboard/hooks/useTimeseries.ts [hunk 6]
```diff
@@ -1,14 +1,25 @@
+import { useEffect, useState, SetStateAction, Dispatch } from 'react';
 import Client from '@hiro-graph/client';
-import { useEffect, useState } from 'react';
+
+interface TS {
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/hooks/useTimeseries.ts]: find a better name

File: packages/bootstrap/src/components/dashboard/types.ts [hunk 7]
```diff
@@ -1,4 +1,8 @@
-import { HiroPartialTheme } from '../../types';
+import {
+  ColorDefinitionValuesEnum,
+  HiroColors,
+  HiroPartialTheme,
+} from '../../types';
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/types.ts]: import order

File: packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+  { from, to }: { from: number; to: number },
+) {
+  return useTimeseries(client, name, {
+    from: from,
+    to: to,
+    limit: 1,
+  });
+}
+
+export const StatisticsController: React.FC = () => {
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/widgets/Statistic/StatisticsController.tsx]: This component looks messy. Can we split data loading into separate hooks?

File: packages/bootstrap/src/components/dashboard/hooks/useTimeseries.ts (hunk 5 above)
Comment (vteremasov) [reply packages/bootstrap/src/components/dashboard/hooks/useTimeseries.ts]: Because it was incorrect. Now it's correct

Comment (miraage) [reply packages/bootstrap/src/components/dashboard/hooks/useTimeseries.ts]: Ah, got it

**Pull request merged**